pkg/odo/cli/component/common_link.go

```diff
@@ -446,32 +426,33 @@ func (o *commonLinkOptions) validateForOperator() (err error) {
 // the current component with the given component's service
 // and stores the link info in the env
 func (o *commonLinkOptions) linkOperator() (err error) {
-	// convert service binding request into a map[string]interface{} type so
-	// as to use it with dynamic client
-	serviceBindingMap := make(map[string]interface{})
+	// Convert ServiceBinding -> JSON -> Map -> YAML
+	// JSON conversion step is necessary to inline TypeMeta
+
 	intermediate, err := json.Marshal(o.serviceBinding)
 	if err != nil {
 		return err
 	}
+
+	serviceBindingMap := make(map[string]interface{})
 	err = json.Unmarshal(intermediate, &serviceBindingMap)
 	if err != nil {
 		return err
 	}
 
-	// this creates a link by creating a service of type
-	// "ServiceBindingRequest" from the Operator "ServiceBindingOperator".
-	err = o.KClient.CreateDynamicResource(serviceBindingMap, kclient.ServiceBindingGroup, kclient.ServiceBindingVersion, kclient.ServiceBindingResource)
+	yamlDesc, err := yaml.Marshal(serviceBindingMap)
 	if err != nil {
-		if strings.Contains(err.Error(), "already exists") {
-			return fmt.Errorf("component %q is already linked with the service %q", o.Context.EnvSpecificInfo.GetName(), o.suppliedName)
-		}
 		return err
 	}
 
-	// once the link is created, we need to store the information in
-	// env.yaml so that subsequent odo push can create a new deployment
-	// based on it
-	err = o.Context.EnvSpecificInfo.SetConfiguration("link", envinfo.EnvInfoLink{Name: o.serviceBinding.GetName(), ServiceKind: o.serviceType, ServiceName: o.serviceName})
+	_, found, err := svc.FindDevfileServiceBinding(o.EnvSpecificInfo.GetDevfileObj(), o.serviceType, o.serviceName)
+	if err != nil {
+		return err
+	}
+	if found {
+		return fmt.Errorf("component %q is already linked with the service %q", o.Context.EnvSpecificInfo.GetName(), o.suppliedName)
```

Comment:
It should be `already linked with the component` in case of a linked component
```
$ odo link frontend
 ✓  Successfully created link between component "nodejs-project-lagl" and component "frontend"

To apply the link, please use `odo push`
$ odo link frontend
 ✗  component "nodejs-project-lagl" is already linked with the service "frontend"
```

Reply:
done